macros/scripts/audio.asm

```diff
@@ -25,24 +46,33 @@ FIRST_MUSIC_CMD EQU __enum__
 
 	enum octave_cmd ; $d0
 octave: MACRO
-	db octave_cmd | 8 - (\1)
+	db octave_cmd | 8 - (\1) ; octave
 ENDM
 
 __enumdir__ = +1
 
-	enum notetype_cmd ; $d8
-notetype: MACRO
-	db notetype_cmd
-	db \1 ; note_length
-if _NARG >= 2
-	db \2 ; intensity
-endc
+	enum note_type_cmd ; $d8
+note_type: MACRO
+	db note_type_cmd
+	db \1 ; note length
+	IF _NARG >= 2
+		IF \3 < 0
+			db (\2 << 4) | (%1000 | (\3 * -1)) ; volume envelope
+		ELSE
+			db (\2 << 4) | \3 ; volume envelope
+		ENDC
+	ENDC
 ENDM
 
-	enum pitchoffset_cmd ; $d9
-pitchoffset: MACRO
-	db pitchoffset_cmd
-	dn \1, \2 - 1 ; octave, key
+; only valid on the noise channel
+dspeed: MACRO
```

Comment:
`drum_speed`, and technically redundant with `note_type`, since they both are declaring a length. Actually, since `note_type`'s second arg is optional, how about having `note_length` (one arg, any channel) and `note_type` (both args required)?

Reply:
I already explained why `dspeed/drum_speed` deserves to be its own macro. I do not like `note_length` because it implies it works the same on any channel. `dspeed` is only valid on the noise channel.

Reply:
Changed to `drum_speed`